### DISCUSSION TITLE
Bold Reports in Oqtane module

### DISCUSSION THREAD

**Discussion by LuisVilaplana** [oqtane/oqtane.framework#4285] [General]
Hello, has anyone been able to get Bold Reports to work in an Oqtane module? I have followed all the possible guides and I can't get it to work, I always get this error:

Microsoft.JSInterop.JSException: '$(...).boldReportViewer is not a function
RenderViewer@http://localhost:44357/reporte line 1 > injectedScript:5:32
beginInvokeJSFromDotNet/i<@http://localhost:44357/_framework/blazor.web.js:1:3244
beginInvokeJSFromDotNet@http://localhost:44357/_framework/blazor.web.js:1:3201

It works fine in regular Blazor server applications, but not inside a module. These are the resources I'm using:

@code {
    public override List<Resource> Resources => new List<Resource>()
    {
        new Resource { ResourceType = ResourceType.Stylesheet, Url = "_content/Syncfusion.Blazor/styles/bootstrap5.css" },
        new Resource { ResourceType = ResourceType.Stylesheet, Url = "https://cdn.boldreports.com/6.1.36/content/material/bold.reports.all.min.css" },
        new Resource { ResourceType = ResourceType.Stylesheet, Url = "https://cdn.boldreports.com/6.1.36/content/material/bold.theme.compatibility.min.css" },
        new Resource { ResourceType = ResourceType.Script, Url = "https://cdnjs.cloudflare.com/ajax/libs/jquery/3.6.0/jquery.min.js" },
        new Resource { ResourceType = ResourceType.Script, Url = "https://cdn.boldreports.com/6.1.36/scripts/common/bold.reports.common.min.js" },
        new Resource { ResourceType = ResourceType.Script, Url = "https://cdn.boldreports.com/6.1.36/scripts/common/bold.reports.widgets.min.js" },
        new Resource { ResourceType = ResourceType.Script, Url = "https://cdn.boldreports.com/6.1.36/scripts/bold.report-viewer.min.js" },
        new Resource { ResourceType = ResourceType.Script, Url = "~/scripts/boldreports-interop.js" },
    };
...

Thanks in advance

**Comment by leigh-pointer** [comment]
@LuisVilaplana it looks like when building the external resources are not available. If you modify you debug.cmd fill to copy from your client and server project to the correct location in your oqtane.server project. Also make sure your js css files are also registered correctly.

**Reply by LuisVilaplana** [reply]
Thank you very much, @leigh-pointer. I did a double-check, and it seems that all the necessary DLLs and JS files are copied. Even for the report viewer, I referenced them directly from the CDN to avoid issues. According to this document, several of these references actually need to be deleted for version 2.0, but it still doesn't work.

https://help.boldreports.com/report-viewer-sdk/javascript-reporting/report-viewer/how-to/migrate-report-viewer-v2/

Originally I followed this guide and others I found as well:

https://help.boldreports.com/report-viewer-sdk/blazor-reporting/report-viewer/add-report-viewer-to-a-blazor-application/

**Reply by LuisVilaplana** [reply]
I did a simple test: I removed all references to Syncfusion and Bold Reports in my module and deleted all the static files from wwwroot that were being copied with debug.cmd. I installed the Syncfusion NuGet package in the Oqtane.Framework project (I know this is not good practice and goes against Oqtane's philosophy, it's just a test) and all Syncfusion components work well in my modules except for the PDF viewer (SfPdfViewer) and the Bold Reports viewer. Could th
ere be something in Oqtane that prevents these components from working?

**Reply by leigh-pointer** [reply]
PDF issue might be due to this?
![image](https://github.com/oqtane/oqtane.framework/assets/9447612/fc3445f1-97a7-4f24-ae85-b6b6237c8bcc)

**Reply by leigh-pointer** [reply]
@LuisVilaplana I have downloaded BolderReports and will try and get it to work for you.

**Reply by LuisVilaplana** [reply]
Thank you very much, I really appreciate your help, it is very important for me to be able to make these two components work because otherwise I will have to abandon Oqtane and I would not want to do that. I'm going to continue trying to see if I can make progress.
Regards

**Reply by LuisVilaplana** [reply]
For now I have managed to read the rdlc report with the Bold Reports Writer and convert it to pdf from a memorystream using the browser's pdf viewer:

 <iframe src="@documentPath" width="100%" height="100%" frameborder="0" type="application/pdf"></iframe> 


It works well and fast but is more difficult to customize
